### PR TITLE
GEOMESA-1355 Migrate some Accumulo utilities to general utilities.

### PR DESCRIPTION
**Pull request by aheyne**
Moved
org.locationtech.geomesa.accumulo.util.UniqueMultiCollection
org.locationtech.geomesa.accumulo.util.Z3FeatureIdGenerator
org.locationtech.geomesa.accumulo.util.Z3UuidGenerator
from geomesa.accumulo.util to geomesa.utils and updated references.

Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

File: geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/CartesianProductIterableTest.scala [hunk 1]
```diff
@@ -9,6 +9,7 @@
 package org.locationtech.geomesa.utils
```
Comment (elahrvivaz) [inline geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/CartesianProductIterableTest.scala]: Move this into the 'iterators' package so it matches the source class

File: geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/InfiniteIteratorTest.scala [hunk 2]
```diff
@@ -18,7 +18,7 @@ class InfiniteIteratorTest extends Specification with Mockito {
 
   "stop after" should {
 
-    import InfiniteIterator._
+    import org.locationtech.geomesa.utils.iterators.InfiniteIterator._
```
Comment (elahrvivaz) [inline geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/InfiniteIteratorTest.scala]: move this into the 'iterators' package too

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/Z3FeatureIdGenerator.scala [hunk 3]
```diff
@@ -6,7 +6,7 @@
 * http://www.opensource.org/licenses/apache2.0.php.
 *************************************************************************/
 
-package org.locationtech.geomesa.accumulo.util
+package org.locationtech.geomesa.utils
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/Z3FeatureIdGenerator.scala]: let's move this into the 'uuid' package (next to the interface)

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/UniqueMultiCollection.scala [hunk 4]
```diff
@@ -6,7 +6,7 @@
 * http://www.opensource.org/licenses/apache2.0.php.
 *************************************************************************/
 
-package org.locationtech.geomesa.accumulo.util
+package org.locationtech.geomesa.utils
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/UniqueMultiCollection.scala]: let's move this to the 'geotools' package

**Comment by elahrvivaz** [comment]
looks good. does it build? i noticed some imports got removed that might still be needed.

**Comment by aheyne** [comment]
No, the build just failed on me... give me 10 mins.

**Comment by aheyne** [comment]
This should be good to go now.

**Comment by elahrvivaz** [comment]
:shipit:

**Pull request merged**